script/tool/lib/src/batch_command.dart

```diff
@@ -0,0 +1,339 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' as io;
+import 'dart:math' as math;
+
+import 'package:file/file.dart';
+import 'package:git/git.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:yaml/yaml.dart';
+import 'package:yaml_edit/yaml_edit.dart';
+
+import 'common/core.dart';
+import 'common/output_utils.dart';
+import 'common/package_command.dart';
+import 'common/repository_package.dart';
+
+const int _kExitPackageMalformed = 2;
+const int _kGitFailedToPush = 3;
+
+const String _kTemplateFileName = 'template.yaml';
+
+/// A command to create a pull request for a single package release.
+class BatchCommand extends PackageCommand {
+  /// Creates a new `batch` command.
+  BatchCommand(
+    super.packagesDir, {
+    super.processRunner,
+    super.platform,
+    super.gitDir,
+  }) {
+    argParser.addOption(
+      'package',
```

Comment:
Commands should not have two different ways to specify what they target. `PackageCommand` already has the ability to specify target packages.